docker/apache/sites-available/ssl-courtlistener.com.conf

```diff
@@ -9,13 +11,11 @@
     # Since Let's Encrypt can auto-update some of this stuff it just seems
     # safer to repeat it.
     SSLEngine on
-    SSLProtocol all -SSLv2 -SSLv3
-    SSLCipherSuite "kEECDH+ECDSA+AES128 kEECDH+ECDSA+AES256 kEECDH+AES128 kEECDH+AES256 kEDH+AES128 kEDH+AES256 DES-CBC3-SHA +SHA !aNULL !eNULL !LOW !MD5 !EXP !DSS !PSK !SRP !kECDH !CAMELLIA !RC4 !SEED"
-    SSLHonorCipherOrder on
+    SSLProtocol            all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
```

Comment:
Do you still need to support SSLv2? It's not recommended due to downgrade attacks and other security flaws.

p.s. have you seen Mozilla tool for up-to-date secure configs for various servers? https://ssl-config.mozilla.org/#server=apache&version=2.4.41&config=intermediate&openssl=1.1.1d&guideline=5.6

Reply:
The other thing that the Mozilla tool mentions is setting `SSLSessionTickets off` which ([apparently](https://httpd.apache.org/docs/trunk/ssl/ssl_howto.html)) ensures forward secrecy.

There's also OCSP stapling which seems to mean that the browser doesn't need to contact the CA to check that the certificate hasn't been revoked. Enabling stapling can reduce the TLS overhead by about a third ([apparently](https://blog.cloudflare.com/ocsp-stapling-how-cloudflare-just-made-ssl-30/)).

Reply:
> `SSLSessionTickets`

I used the mozilla configurator, but missed the `SSLSessionTickets` directive. Thanks for catching that. Fixed.

> `SSLv2`

The "Intermediate" setting for the configurator doesn't ban SSLv2, so I'm just rolling with that. That said, once this is up, I think I'll point the SSLLabs test thing at it, and if it tells me to block more protocols, I probably will.

> OCSP stapling

I don't know much about OCSP stapling. I'm in favor of it generally, but I haven't done the research to learn if it has any downsides. Doesn't seem like it? That CloudFlare thing is eight years old! Do you know though if it's just safe to turn it on with no downsides?

Reply:
Both the "intermediate" and "old" settings leave out SSLv2 when I look at it.

I'm not sure whether OCSP stapling has any downsides tbh, I just found out about it from that Mozilla tool and some Googling around. It defaults to ticked in the SSL configuration generator, anyway.